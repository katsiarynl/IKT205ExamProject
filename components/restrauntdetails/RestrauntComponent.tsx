import { View, Image } from "react-native";
import React, { useState } from "react";

import restrauntInfoStyles from "../../styles/restrauntinfostyles";
import RestrauntDescriptionComponent from "./RestrauntDescriptionComponent";

import fetchImage from "../../utilities/FetchImage";
import GETPath from "../../utilities/GETPath";
import { RestaurantComponentType } from "../../types/restrauntTypes";
export default function RestrauntComponent(props: RestaurantComponentType) {
  const [avatare, setAvatar] = useState("");
  fetchImage(GETPath, setAvatar, props.image);
  return (
    <View style={restrauntInfoStyles.restrauntview}>
      <Image
        style={{
          width: "100%",
          height: "100%",
          marginBottom: "-5%",
          flex: 2,
        }}
        source={{
          uri: avatare,
        }}
      />
      <RestrauntDescriptionComponent
        name={props.name}
        description={props.description}
      />
    </View>
  );
}
