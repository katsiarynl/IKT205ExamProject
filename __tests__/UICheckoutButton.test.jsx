import React from "react";
import { create } from "react-test-renderer";

import RestrauntComponent from "../components/restrauntdetails/RestrauntComponent";
const tree = create(<RestrauntComponent />);

test("snapshot", () => {
  expect(tree).toMatchSnapshot();
});
