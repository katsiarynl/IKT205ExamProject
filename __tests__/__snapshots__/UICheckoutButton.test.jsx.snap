// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot 1`] = `
<View
  style={
    {
      "height": 533.6,
      "margin": -5,
      "paddingTop": "13%",
    }
  }
>
  <Image
    source={
      {
        "uri": "",
      }
    }
    style={
      {
        "flex": 2,
        "height": "100%",
        "marginBottom": "-5%",
        "width": "100%",
      }
    }
  />
  <View
    style={
      {
        "backgroundColor": "#ffff",
        "borderTopLeftRadius": 40,
        "borderTopRightRadius": 40,
        "flex": 1,
      }
    }
  >
    <Text>
      RestrauntDescriptionComponent
    </Text>
  </View>
</View>
`;
